pkg/apis/kubeadm/bootstraptoken/v1/utils.go

```diff
@@ -71,8 +81,7 @@ func (bts BootstrapTokenString) String() string {
 // is of the right format
 func NewBootstrapTokenString(token string) (*BootstrapTokenString, error) {
 	substrs := bootstraputil.BootstrapTokenRegexp.FindStringSubmatch(token)
-	// TODO: Add a constant for the 3 value here, and explain better why it's needed (other than because how the regexp parsing works)
-	if len(substrs) != 3 {
+	if len(substrs) != validatedSubstringsSize {
```

Comment:
You can simply `if substrs == nil` to see if the regex matched.

Reply:
This file is a copy from k/k, I'd rather leave it as it is to make keeping it up to date easier.